src/framework/audio/internal/platform/lin/linuxaudiodriver.h

```diff
@@ -71,6 +82,12 @@ class LinuxAudioDriver : public IAudioDriver, public async::Asyncable
 
     async::Notification m_bufferSizeChanged;
     async::Notification m_sampleRateChanged;
+    int m_audioDelayCompensate;
+
+    struct IAudioDriver::Spec m_spec;
+    std::unique_ptr<AudioDriverState> m_current_audioDriverState;
```

Comment:
Sorry for talking about this so late.. 
I think it’s better to remove `LinuxAudioDriver ` and keep only `AlsaAudioDriver` and `JackAudioDriver` as two independent implementations. To switch between them, you need to add another class, something like `IAudioDriverProvider`. This is what you need to add to IoC (and remove the drivers from IoC). Access the driver through this class, like: 

```
class IAudioDriverProvider ...
{
...

        std::shared_ptr<IAudioDriver> driver() const;
  
        void swithDriver(const std::string& name); 
        async::Chanel<std::string /*name*/> dirverChanged() const;
}
```


```
class SomeClass 
    Inject<IAudioDriverProvider > audioDriverProvider;


...

     audioDriverProvider()->driver();
```


This greatly simplifies everything, each class becomes simple and does one thing.
Moreover, such a system is easily scalable; we can easily add other drivers, cross-platform or platform-specific. 

But I understand that this should have been proposed a long time ago... Therefore, I don't insist, it can be reimplemented later..

Now I mainly don't like passing modules classes into each other and further down into a bunch of classes... this is definitely not necessary :)